projects/plugins/jetpack/functions.photon.php

```diff
@@ -3,339 +3,79 @@
-		|| wp_parse_url( $custom_photon_url, PHP_URL_HOST ) === $image_url_parts['host']
-		|| $is_wpcom_image
-	) {
-		$photon_url = add_query_arg( $args, $image_url );
-		return jetpack_photon_url_scheme( $photon_url, $scheme );
-	}
-
-	/**
-	 * Allow Photon to use query strings as well.
-	 * By default, Photon doesn't support query strings so we ignore them and look only at the path.
-	 * This setting is Photon Server dependent.
-	 *
-	 * @module photon
-	 *
-	 * @since 1.9.0
-	 *
-	 * @param bool false Should images using query strings go through Photon. Default is false.
-	 * @param string $image_url_parts['host'] Image URL's host.
-	 */
-	if ( ! apply_filters( 'jetpack_photon_any_extension_for_domain', false, $image_url_parts['host'] ) ) {
-		// Photon doesn't support query strings so we ignore them and look only at the path.
-		// However some source images are served via PHP so check the no-query-string extension.
-		// For future proofing, this is an excluded list of common issues rather than an allow list.
-		$extension = pathinfo( $image_url_parts['path'], PATHINFO_EXTENSION );
-		if ( empty( $extension ) || in_array( $extension, array( 'php', 'ashx' ), true ) ) {
-			return $image_url;
-		}
-	}
-
-	$image_host_path = $image_url_parts['host'] . $image_url_parts['path'];
-
-	/**
-	 * Filters the domain used by the Photon module.
-	 *
-	 * @module photon
-	 *
-	 * @since 3.4.2
-	 *
-	 * @param string https://i0.wp.com Domain used by Photon.
-	 * @param string $image_url URL of the image to be photonized.
-	 */
-	$photon_domain = apply_filters( 'jetpack_photon_domain', 'https://i0.wp.com', $image_url );
-	$photon_domain = trailingslashit( esc_url( $photon_domain ) );
-	$photon_url    = $photon_domain . $image_host_path;
-
-	/**
-	 * Add query strings to Photon URL.
-	 * By default, Photon doesn't support query strings so we ignore them.
-	 * This setting is Photon Server dependent.
-	 *
-	 * @module photon
-	 *
-	 * @since 1.9.0
-	 *
-	 * @param bool false Should query strings be added to the image URL. Default is false.
-	 * @param string $image_url_parts['host'] Image URL's host.
-	 */
-	if ( isset( $image_url_parts['query'] ) && apply_filters( 'jetpack_photon_add_query_string_to_domain', false, $image_url_parts['host'] ) ) {
-		$photon_url .= '?q=' . rawurlencode( $image_url_parts['query'] );
-	}
-
-	if ( $args ) {
-		if ( is_array( $args ) ) {
-			$photon_url = add_query_arg( $args, $photon_url );
-		} else {
-			// You can pass a query string for complicated requests but where you still want CDN subdomain help, etc.
-			$photon_url .= '?' . $args;
-		}
-	}
-
-	if ( isset( $image_url_parts['scheme'] ) && 'https' === $image_url_parts['scheme'] ) {
-		$photon_url = add_query_arg( array( 'ssl' => 1 ), $photon_url );
-	}
-
-	return jetpack_photon_url_scheme( $photon_url, $scheme );
+	_deprecated_function( __FUNCTION__, 'jetpack-$$next-version$$', 'Automattic\Jetpack\Image_CDN\Image_CDN_Core::cdn_url' );
```

Comment:
Coming back to this, I have mixed feelings about this.

- On one hand, we want plugin and theme authors to start using the new package instead of the legacy feature.
- On the other hand, there are many themes and plugins using the function today. Some of those themes have more than 30,000 installs. This deprecation notice, while only visible in the logs on most sites, will generate some support questions for Jetpack and for those theme authors. It will even generate notices on WordPress.com where we use `jetpack_photon_url` in many places.

With that in mind, I'm thinking we may want to keep this specific function around without a deprecation notice, as an exception.

What do you think?

Reply:
I agree, and this has concerned me as well. Since we have the depreciation marked in the doc block, it should be alright. I'll address it in a separate PR.